src/modules/referral/services/cron-service.ts

```diff
@@ -3,43 +3,62 @@ import { Repository } from "typeorm";
 import { InjectRepository } from "@nestjs/typeorm";
 import { AppConfig } from "../../configuration/configuration.service";
 import { Deposit } from "../../scraper/model/deposit.entity";
-import { EnhancedCron } from "../../../utils";
-import { updateStickyReferralAddresses } from "../services/queries";
+import { EnhancedCron, wait } from "../../../utils";
+import { updateStickyReferralAddressesQuery } from "../services/queries";
+import { StickyReferralAddressesMechanism } from "src/modules/configuration";
 
 @Injectable()
 export class ReferralCronService {
   private logger = new Logger(ReferralCronService.name);
+  private semaphore = false;
 
   constructor(
     @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
     private appConfig: AppConfig,
   ) {}
 
-  // run cron every 8 minutes
-  @EnhancedCron("0 4-59/8 * * * *")
-  async refreshMaterializedViewCron() {
-    this.logger.log("start refreshMaterializedViewCron()");
-    if (this.appConfig.values.enableReferralsMaterializedViewRefresh) {
+  @EnhancedCron("0 */10 * * * *")
+  async startCrons() {
```

Comment:
Is it possible something in this method throws and then the semaphore never gets reset?

Reply:
Great observation 👍 
No, theoretically this method should not throw errors, but I wrapped everything in a `try catch finally` block. This will make it less error prone in case of future changes. 

New version ⬇️

https://github.com/across-protocol/scraper-api/blob/109e9c4ff4c641687e3ae8eade4cf88562f2bd73/src/modules/referral/services/cron-service.ts#L22-L34